src/main/groovy/lang/ObjectRange.java

```diff
@@ -240,16 +237,16 @@ public Object get(int index) {
         if (index < 0) {
             throw new IndexOutOfBoundsException("Index: " + index + " should not be negative");
         }
-        final StepIterator iter = new StepIterator(this, 1);
 
-        Object value = iter.next();
-        for (int i = 0; i < index; i++) {
-            if (!iter.hasNext()) {
+        int i = 0;
+        for (Iterator it = iterator(); ; i++) {
```

Comment:
and generally it is unclear to me why you changed the loop style here. I am not sure the new structure is objectively better than before.